### PR TITLE
Add uthash

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Add uthash to obs-deps.

Yes, they should be installed in the include directory without any subdirectory to align with Linux distros.

### Motivation and Context

Reduce in-tree vendored deps in OBS Studio repo.

### How Has This Been Tested?

CI builds artifacts on my fork PR, they contain headers and license files.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by derrod** [comment]
Is there a specific reason to include the other headers? We only need `uthash.h` and there should be no interdependence between them.

**Comment by tytan652** [comment]
I just did it like a packager should do it since all the header are part of uthash.

File: deps.windows/60-uthash.ps1 [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+function Install {
+    Log-Information "Install (${Target})"
+    Set-Location "$Path"
+
+    New-Item -Path "$($ConfigData.OutputPath)/include" -ItemType Directory -Force *> $null
+
+    $Items = @(
+        @{
+            Path = "src/utarray.h"
```
Comment (PatTheMav) [inline deps.windows/60-uthash.ps1:20]: You should be able to specify an array here (`"src/utarray.h", "src/uthash.h", [..]`), PowerShell usually accepts that as input parameter for file operations.

Comment (tytan652) [reply deps.windows/60-uthash.ps1]: No log is fine, or do you want some kind of logging ?

I ask before I resolve this and squash my commits.

Comment (PatTheMav) [reply deps.windows/60-uthash.ps1]: Why did you remove the rest of the code? That code was already compatible with providing an array (hence why it did the whole thing about joining the string). The only change necessary was to have a single item with the files provided as an array. :D

Comment (tytan652) [reply deps.windows/60-uthash.ps1]: You say to use an array which usually make the whole for loop useless (and so the log per file).

Comment (tytan652) [reply deps.windows/60-uthash.ps1]: So you want to keep the loop ?

**Review by PatTheMav — APPROVED** [review]
Should be good. I also agree with copying all related files to make it a "default" distribution.

The compiler will pull in files as needed.

**Review by Conan-Kudo — APPROVED** [review]

**Review by RytoEX — APPROVED** [review]
Seems fine at a glance. Will merge when we get to a merge window.

**Pull request merged**